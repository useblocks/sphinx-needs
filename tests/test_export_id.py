import json
import os
from pathlib import Path

import pytest
from syrupy.filters import props


@pytest.mark.parametrize("test_app", [{"buildername": "needs", "srcdir": "doc_test/doc_export_id"}], indirect=True)
def test_export_id(test_app, snapshot):
    app = test_app
    app.build()
    needs_data = json.loads(Path(app.outdir, "needs.json").read_text())
    assert needs_data == snapshot(exclude=props("created"))


@pytest.mark.parametrize("test_app", [{"buildername": "html", "srcdir": "doc_test/doc_export_id"}], indirect=True)
def test_export_id_html(test_app):
    app = test_app
    app.build()
    assert not os.path.exists(os.path.join(app.outdir, "needs.json"))
