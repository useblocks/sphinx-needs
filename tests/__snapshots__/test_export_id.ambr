# serializer version: 1
# name: test_export_id[test_app0]
  dict({
    'current_version': '1.0',
    'project': 'Python',
    'versions': dict({
      '1.0': dict({
        'filters': dict({
          'FLOW_1': dict({
            'amount': 10,
            'export_id': 'FLOW_1',
            'filter': '',
            'result': list([
              'REQ_001',
              'REQ_002',
              'REQ_003',
              'REQ_004',
              'REQ_005',
              'TEST_001',
              'TEST_002',
              'TEST_003',
              'REQ_005.1',
              'REQ_005.cool',
            ]),
            'status': list([
            ]),
            'tags': list([
            ]),
            'types': list([
            ]),
          }),
          'FLOW_2': dict({
            'amount': 5,
            'export_id': 'FLOW_2',
            'filter': 'is_need is False or type != "story"',
            'result': list([
              'TEST_001',
              'TEST_002',
              'TEST_003',
              'REQ_005.1',
              'REQ_005.cool',
            ]),
            'status': list([
            ]),
            'tags': list([
            ]),
            'types': list([
            ]),
          }),
          'LIST_1': dict({
            'amount': 10,
            'export_id': 'LIST_1',
            'filter': '',
            'result': list([
              'REQ_001',
              'REQ_002',
              'REQ_003',
              'REQ_004',
              'REQ_005',
              'TEST_001',
              'TEST_002',
              'TEST_003',
              'REQ_005.1',
              'REQ_005.cool',
            ]),
            'status': list([
            ]),
            'tags': list([
            ]),
            'types': list([
            ]),
          }),
          'TABLE_1': dict({
            'amount': 10,
            'export_id': 'TABLE_1',
            'filter': '',
            'result': list([
              'REQ_001',
              'REQ_002',
              'REQ_003',
              'REQ_004',
              'REQ_005',
              'TEST_001',
              'TEST_002',
              'TEST_003',
              'REQ_005.1',
              'REQ_005.cool',
            ]),
            'status': list([
            ]),
            'tags': list([
            ]),
            'types': list([
            ]),
          }),
          'TABLE_2': dict({
            'amount': 3,
            'export_id': 'TABLE_2',
            'filter': '"test" in type',
            'result': list([
              'TEST_001',
              'TEST_002',
              'TEST_003',
            ]),
            'status': list([
            ]),
            'tags': list([
            ]),
            'types': list([
            ]),
          }),
        }),
        'filters_amount': 5,
        'needs': dict({
          'REQ_001': dict({
            'arch': dict({
            }),
            'avatar': '',
            'blocks': list([
              'REQ_003',
            ]),
            'closed_at': '',
            'completion': '',
            'constraints': list([
            ]),
            'constraints_passed': True,
            'constraints_results': dict({
            }),
            'content_id': 'REQ_001',
            'created_at': '',
            'delete': None,
            'description': '',
            'docname': 'index',
            'doctype': '.rst',
            'duration': '',
            'external_css': 'external_link',
            'external_url': None,
            'full_title': 'My requirement',
            'has_dead_links': '',
            'has_forbidden_dead_links': '',
            'hidden': '',
            'id': 'REQ_001',
            'id_prefix': '',
            'is_external': False,
            'is_modified': False,
            'is_need': True,
            'is_part': False,
            'jinja_content': None,
            'layout': '',
            'links': list([
              'REQ_002',
              'REQ_004',
              'REQ_003',
            ]),
            'max_amount': '',
            'max_content_lines': '',
            'modifications': 0,
            'params': '',
            'parent_need': '',
            'parent_needs': list([
            ]),
            'parts': dict({
            }),
            'post_template': None,
            'pre_template': None,
            'prefix': '',
            'query': '',
            'section_name': 'Stories',
            'sections': list([
              'Stories',
              'EXTRA LINKS DOCUMENT',
            ]),
            'service': '',
            'signature': '',
            'specific': '',
            'status': None,
            'style': None,
            'tags': list([
            ]),
            'target_id': 'REQ_001',
            'template': None,
            'tests': list([
            ]),
            'title': 'My requirement',
            'type': 'story',
            'type_name': 'User Story',
            'updated_at': '',
            'url': '',
            'url_postfix': '',
            'user': '',
          }),
          'REQ_002': dict({
            'arch': dict({
            }),
            'avatar': '',
            'blocks': list([
            ]),
            'closed_at': '',
            'completion': '',
            'constraints': list([
            ]),
            'constraints_passed': True,
            'constraints_results': dict({
            }),
            'content_id': 'REQ_002',
            'created_at': '',
            'delete': None,
            'description': '',
            'docname': 'index',
            'doctype': '.rst',
            'duration': '',
            'external_css': 'external_link',
            'external_url': None,
            'full_title': 'My requirement 2',
            'has_dead_links': '',
            'has_forbidden_dead_links': '',
            'hidden': '',
            'id': 'REQ_002',
            'id_prefix': '',
            'is_external': False,
            'is_modified': False,
            'is_need': True,
            'is_part': False,
            'jinja_content': None,
            'layout': '',
            'links': list([
            ]),
            'max_amount': '',
            'max_content_lines': '',
            'modifications': 0,
            'params': '',
            'parent_need': '',
            'parent_needs': list([
            ]),
            'parts': dict({
            }),
            'post_template': None,
            'pre_template': None,
            'prefix': '',
            'query': '',
            'section_name': 'Stories',
            'sections': list([
              'Stories',
              'EXTRA LINKS DOCUMENT',
            ]),
            'service': '',
            'signature': '',
            'specific': '',
            'status': None,
            'style': None,
            'tags': list([
            ]),
            'target_id': 'REQ_002',
            'template': None,
            'tests': list([
            ]),
            'title': 'My requirement 2',
            'type': 'story',
            'type_name': 'User Story',
            'updated_at': '',
            'url': '',
            'url_postfix': '',
            'user': '',
          }),
          'REQ_003': dict({
            'arch': dict({
            }),
            'avatar': '',
            'blocks': list([
            ]),
            'closed_at': '',
            'completion': '',
            'constraints': list([
            ]),
            'constraints_passed': True,
            'constraints_results': dict({
            }),
            'content_id': 'REQ_003',
            'created_at': '',
            'delete': None,
            'description': '',
            'docname': 'index',
            'doctype': '.rst',
            'duration': '',
            'external_css': 'external_link',
            'external_url': None,
            'full_title': 'My requirement 3',
            'has_dead_links': '',
            'has_forbidden_dead_links': '',
            'hidden': '',
            'id': 'REQ_003',
            'id_prefix': '',
            'is_external': False,
            'is_modified': False,
            'is_need': True,
            'is_part': False,
            'jinja_content': None,
            'layout': '',
            'links': list([
            ]),
            'max_amount': '',
            'max_content_lines': '',
            'modifications': 0,
            'params': '',
            'parent_need': '',
            'parent_needs': list([
            ]),
            'parts': dict({
            }),
            'post_template': None,
            'pre_template': None,
            'prefix': '',
            'query': '',
            'section_name': 'Stories',
            'sections': list([
              'Stories',
              'EXTRA LINKS DOCUMENT',
            ]),
            'service': '',
            'signature': '',
            'specific': '',
            'status': None,
            'style': None,
            'tags': list([
            ]),
            'target_id': 'REQ_003',
            'template': None,
            'tests': list([
            ]),
            'title': 'My requirement 3',
            'type': 'story',
            'type_name': 'User Story',
            'updated_at': '',
            'url': '',
            'url_postfix': '',
            'user': '',
          }),
          'REQ_004': dict({
            'arch': dict({
            }),
            'avatar': '',
            'blocks': list([
            ]),
            'closed_at': '',
            'completion': '',
            'constraints': list([
            ]),
            'constraints_passed': True,
            'constraints_results': dict({
            }),
            'content_id': 'REQ_004',
            'created_at': '',
            'delete': None,
            'description': '',
            'docname': 'index',
            'doctype': '.rst',
            'duration': '',
            'external_css': 'external_link',
            'external_url': None,
            'full_title': 'My requirement 4',
            'has_dead_links': '',
            'has_forbidden_dead_links': '',
            'hidden': '',
            'id': 'REQ_004',
            'id_prefix': '',
            'is_external': False,
            'is_modified': False,
            'is_need': True,
            'is_part': False,
            'jinja_content': None,
            'layout': '',
            'links': list([
            ]),
            'max_amount': '',
            'max_content_lines': '',
            'modifications': 0,
            'params': '',
            'parent_need': '',
            'parent_needs': list([
            ]),
            'parts': dict({
            }),
            'post_template': None,
            'pre_template': None,
            'prefix': '',
            'query': '',
            'section_name': 'Stories',
            'sections': list([
              'Stories',
              'EXTRA LINKS DOCUMENT',
            ]),
            'service': '',
            'signature': '',
            'specific': '',
            'status': None,
            'style': None,
            'tags': list([
            ]),
            'target_id': 'REQ_004',
            'template': None,
            'tests': list([
            ]),
            'title': 'My requirement 4',
            'type': 'story',
            'type_name': 'User Story',
            'updated_at': '',
            'url': '',
            'url_postfix': '',
            'user': '',
          }),
          'REQ_005': dict({
            'arch': dict({
            }),
            'avatar': '',
            'blocks': list([
              'REQ_001',
            ]),
            'closed_at': '',
            'completion': '',
            'constraints': list([
            ]),
            'constraints_passed': True,
            'constraints_results': dict({
            }),
            'content_id': 'REQ_005',
            'created_at': '',
            'delete': None,
            'description': '''
              :need_part:`(1) awesome part`
              
              :need_part:`(cool) a cool part`
            ''',
            'docname': 'index',
            'doctype': '.rst',
            'duration': '',
            'external_css': 'external_link',
            'external_url': None,
            'full_title': 'Req 5',
            'has_dead_links': '',
            'has_forbidden_dead_links': '',
            'hidden': '',
            'id': 'REQ_005',
            'id_prefix': '',
            'is_external': False,
            'is_modified': False,
            'is_need': True,
            'is_part': False,
            'jinja_content': None,
            'layout': '',
            'links': list([
              'REQ_001',
              'REQ_001',
            ]),
            'max_amount': '',
            'max_content_lines': '',
            'modifications': 0,
            'params': '',
            'parent_need': '',
            'parent_needs': list([
            ]),
            'parts': dict({
              '1': dict({
                'content': ' awesome part',
                'document': 'index',
                'id': '1',
                'is_need': False,
                'is_part': True,
                'links': list([
                ]),
                'links_back': list([
                  'TEST_003',
                ]),
                'tests_back': list([
                  'TEST_003',
                ]),
              }),
              'cool': dict({
                'content': ' a cool part',
                'document': 'index',
                'id': 'cool',
                'is_need': False,
                'is_part': True,
                'links': list([
                ]),
                'links_back': list([
                ]),
                'tests_back': list([
                  'TEST_003',
                ]),
              }),
            }),
            'post_template': None,
            'pre_template': None,
            'prefix': '',
            'query': '',
            'section_name': 'Stories',
            'sections': list([
              'Stories',
              'EXTRA LINKS DOCUMENT',
            ]),
            'service': '',
            'signature': '',
            'specific': '',
            'status': None,
            'style': None,
            'tags': list([
            ]),
            'target_id': 'REQ_005',
            'template': None,
            'tests': list([
            ]),
            'title': 'Req 5',
            'type': 'story',
            'type_name': 'User Story',
            'updated_at': '',
            'url': '',
            'url_postfix': '',
            'user': '',
          }),
          'TEST_001': dict({
            'arch': dict({
            }),
            'avatar': '',
            'blocks': list([
            ]),
            'closed_at': '',
            'completion': '',
            'constraints': list([
            ]),
            'constraints_passed': True,
            'constraints_results': dict({
            }),
            'content_id': 'TEST_001',
            'created_at': '',
            'delete': None,
            'description': '',
            'docname': 'index',
            'doctype': '.rst',
            'duration': '',
            'external_css': 'external_link',
            'external_url': None,
            'full_title': 'Test of requirements',
            'has_dead_links': '',
            'has_forbidden_dead_links': '',
            'hidden': '',
            'id': 'TEST_001',
            'id_prefix': '',
            'is_external': False,
            'is_modified': False,
            'is_need': True,
            'is_part': False,
            'jinja_content': None,
            'layout': '',
            'links': list([
            ]),
            'max_amount': '',
            'max_content_lines': '',
            'modifications': 0,
            'params': '',
            'parent_need': '',
            'parent_needs': list([
            ]),
            'parts': dict({
            }),
            'post_template': None,
            'pre_template': None,
            'prefix': '',
            'query': '',
            'section_name': 'Tests',
            'sections': list([
              'Tests',
              'EXTRA LINKS DOCUMENT',
            ]),
            'service': '',
            'signature': '',
            'specific': '',
            'status': None,
            'style': None,
            'tags': list([
            ]),
            'target_id': 'TEST_001',
            'template': None,
            'tests': list([
              'REQ_001',
              'REQ_003',
            ]),
            'title': 'Test of requirements',
            'type': 'test',
            'type_name': 'Test Case',
            'updated_at': '',
            'url': '',
            'url_postfix': '',
            'user': '',
          }),
          'TEST_002': dict({
            'arch': dict({
            }),
            'avatar': '',
            'blocks': list([
            ]),
            'closed_at': '',
            'completion': '',
            'constraints': list([
            ]),
            'constraints_passed': True,
            'constraints_results': dict({
            }),
            'content_id': 'TEST_002',
            'created_at': '',
            'delete': None,
            'description': '',
            'docname': 'index',
            'doctype': '.rst',
            'duration': '',
            'external_css': 'external_link',
            'external_url': None,
            'full_title': 'Test of requirements2',
            'has_dead_links': '',
            'has_forbidden_dead_links': '',
            'hidden': '',
            'id': 'TEST_002',
            'id_prefix': '',
            'is_external': False,
            'is_modified': False,
            'is_need': True,
            'is_part': False,
            'jinja_content': None,
            'layout': '',
            'links': list([
              'TEST_001',
            ]),
            'max_amount': '',
            'max_content_lines': '',
            'modifications': 0,
            'params': '',
            'parent_need': '',
            'parent_needs': list([
            ]),
            'parts': dict({
            }),
            'post_template': None,
            'pre_template': None,
            'prefix': '',
            'query': '',
            'section_name': 'Tests',
            'sections': list([
              'Tests',
              'EXTRA LINKS DOCUMENT',
            ]),
            'service': '',
            'signature': '',
            'specific': '',
            'status': None,
            'style': None,
            'tags': list([
            ]),
            'target_id': 'TEST_002',
            'template': None,
            'tests': list([
              'REQ_001',
            ]),
            'title': 'Test of requirements2',
            'type': 'test',
            'type_name': 'Test Case',
            'updated_at': '',
            'url': '',
            'url_postfix': '',
            'user': '',
          }),
          'TEST_003': dict({
            'arch': dict({
            }),
            'avatar': '',
            'blocks': list([
            ]),
            'closed_at': '',
            'completion': '',
            'constraints': list([
            ]),
            'constraints_passed': True,
            'constraints_results': dict({
            }),
            'content_id': 'TEST_003',
            'created_at': '',
            'delete': None,
            'description': '',
            'docname': 'index',
            'doctype': '.rst',
            'duration': '',
            'external_css': 'external_link',
            'external_url': None,
            'full_title': 'Test of requirements 5',
            'has_dead_links': '',
            'has_forbidden_dead_links': '',
            'hidden': '',
            'id': 'TEST_003',
            'id_prefix': '',
            'is_external': False,
            'is_modified': False,
            'is_need': True,
            'is_part': False,
            'jinja_content': None,
            'layout': '',
            'links': list([
              'REQ_005.1',
            ]),
            'max_amount': '',
            'max_content_lines': '',
            'modifications': 0,
            'params': '',
            'parent_need': '',
            'parent_needs': list([
            ]),
            'parts': dict({
            }),
            'post_template': None,
            'pre_template': None,
            'prefix': '',
            'query': '',
            'section_name': 'Tests',
            'sections': list([
              'Tests',
              'EXTRA LINKS DOCUMENT',
            ]),
            'service': '',
            'signature': '',
            'specific': '',
            'status': None,
            'style': None,
            'tags': list([
            ]),
            'target_id': 'TEST_003',
            'template': None,
            'tests': list([
              'REQ_005.1',
              'REQ_005.cool',
            ]),
            'title': 'Test of requirements 5',
            'type': 'test',
            'type_name': 'Test Case',
            'updated_at': '',
            'url': '',
            'url_postfix': '',
            'user': '',
          }),
        }),
        'needs_amount': 8,
      }),
    }),
  })
# ---
